### PR TITLE
Sending email functionality

### LINKED ISSUE #69: Email Notification System [ Epic 🏔️ ]
# Overview

Create a system to notify HLs/RS by email when statuses of referrals change or when they have been assigned a referral.

These events should trigger an email notification to the relevant party:

- When HLs get assigned to a referral → notify relevant HL 
- When RS gets elevated to HL → notify the user affected
- When HL gets demoted to RS → notify the user affected
- When referral status(es) for unit is set to leased → notify all HLs associated with the ACTIVE referrals for the unit
    - inactive referral statuses = [ “Denied”, “Canceled” ]

## Tasks
- [ ]

### PR DESCRIPTION
**Pull request by YashRavipati1**
## Tracking Info

Resolves #69 

## Changes

Implemented functionality for automated emails (backend and frontend API service)

## Testing

Created a custom button to send a pre-set automated email to a certain email address every time it was pressed.

## Confirmation of Change

Screenshot 2024-05-07 at 2 19 47 AM

### CODE REVIEW

**Review by petabite — COMMENTED** [review]
Do we need a route to send emails? Since all the events that trigger an email are processed in the backend, I was thinking we can just call the backend email service's `sendEmail` function wherever we need to send emails. That way we can save a round trip from the frontend to backend. Thoughts?

File: frontend/package-lock.json
Comment (petabite) [inline frontend/package-lock.json:1]: Can we delete this file? Looks like you maybe accidentally ran `npm install`

File: backend/src/services/email.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    host: "smtp.gmail.com",
+    auth: {
+      user: process.env.GMAILUSER,
+      pass: process.env.GMAILPASS,
+    },
+    secure: true,
+  });
+
+  const mailData = {
+    from: "dummyushsacc@gmail.com", // sender address
```
Comment (petabite) [inline backend/src/services/email.ts]: Is this the same as `process.env.GMAILUSER`?

File: backend/src/services/referral.ts [hunk 2]
```diff
@@ -33,11 +36,39 @@ export async function editReferral(
   assignedReferringStaffId: string,
   status: string,
 ) {
+  const Ref = await ReferralModel.findById(id);
+  const updateHL = Ref?.assignedHousingLocator?.toString() !== assignedHousingLocatorId;
+  const setLeased =
+    status === "leased" &&
```
Comment (petabite) [inline backend/src/services/referral.ts]:
```suggestion
    status === "Leased" &&
```
Should this be `"Leased"` if we are expecting all statuses to start with a capital letter?

File: backend/src/services/referral.ts [hunk 3]
```diff
@@ -33,11 +36,39 @@ export async function editReferral(
...
+  }
+  if (setLeased) {
+    if (HL !== null) {
+      await sendEmail(
+        HL?.email,
+        "Referral Update",
+        `Your referral has been marked as leased. Please login to the portal to view the update.`,
+      );
+    }
+  }
```
Comment (petabite) [inline backend/src/services/referral.ts]: > - When referral status(es) for unit is set to leased → notify all HLs associated with the ACTIVE referrals for the unit
>     - inactive referral statuses = [ “Denied”, “Canceled” ]

Oops maybe I wasn't too clear about this. When a referral gets set to "Leased", the unit for that referral is leased, so we want to notify all _other_ referrals for that unit that the unit is not available anymore. This currently looks like it just notifies the HL for the current referral.

**Review by petabite — CHANGES_REQUESTED** [review]
Looking great so far! Just a couple things:

**Review by petabite — APPROVED** [review]
💯 🔥

**Review by Azhou2023 — APPROVED** [review]
LGTM!

**Pull request merged**